doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+    from ray.rllib.core.rl_module import RLModuleSpec
+    
+    # Set up a path for the tabular data records.
+    tabular_data_path = "tmp/docs_rllib_offline_recording_tabular"
 
+    # Configure the algorithm for recording.
     config = (
-        DQNConfig()
-        .environment(env="CartPole-v1")
-        .framework("torch")
-        .offline_data(input_="/tmp/cartpole-out")
+        PPOConfig()
+        # The environment needs to be specified.
+        .environment(
+            env="CartPole-v1",
+        )
+        # Make sure you use the new API stack.
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        # Make sure to sample complete episodes because
+        # you want to record RLlib's episode objects.
+        .env_runners(
+            batch_mode="complete_episodes",
+        )
+        # Set up 5 evaluation `EnvRunners` for recording.
+        # Sample 50 episodes in each evaluation rollout.
         .evaluation(
-            evaluation_interval=1,
-            evaluation_duration=10,
-            evaluation_num_env_runners=1,
-            evaluation_duration_unit="episodes",
-            evaluation_config={"input": "/tmp/cartpole-eval"},
-            off_policy_estimation_methods={
-                "is": {"type": ImportanceSampling},
-                "wis": {"type": WeightedImportanceSampling},
-                "dm_fqe": {
-                    "type": DirectMethod,
-                    "q_model_config": {"type": FQETorchModel, "polyak_coef": 0.05},
-                },
-                "dr_fqe": {
-                    "type": DoublyRobust,
-                    "q_model_config": {"type": FQETorchModel, "polyak_coef": 0.05},
-                },
-            },
+            evaluation_num_env_runners=5,
+            evaluation_duration=50,
+        )
+        # Use the checkpointed expert policy from PPO training
+        # above. Note, we have to use the same `model_config` as
+        # the one with which the expert policy was trained, otherwise
+        # the module state cannot be loaded.
+        .rl_module(
+            model_config=DefaultModelConfig(
+                fcnet_hiddens=[32],
+                fcnet_activation="linear",
+                # Share encoder layers between value network
+                # and policy.
+                vf_share_layers=True,
+            ),
+        )
+        # Define the output path and format. In this example you
+        # want to store data directly in RLlib's episode objects. 
+        .offline_data(
+            output=tabular_data_path,
+            # You want to store for this example tabular data.
+            output_write_episodes=False,
         )
     )
 
+    # Build the algorithm.
     algo = config.build()
-    for _ in range(100):
-        algo.train()
+    # Load now the PPO-trained `RLModule` to use in recording.
+    algo.restore_from_path(
+        best_checkpoint,
```

Comment:
same here, correct?

We need to specify the individual RLModule sub-path instead of this algo top-level path.